osu.Framework/Game_Audio.cs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System.Collections.Generic;
+using System.Linq;
+using ManagedBass;
+using ManagedBass.Wasapi;
+using osu.Framework.Audio.Manager.Bass;
+using osu.Framework.Configuration;
+using osu.Framework.IO.Stores;
+using osu.Framework.Logging;
+using osu.Framework.Platform;
+
+namespace osu.Framework
+{
+    public abstract partial class Game
+    {
+        public AudioBackend ResolvedAudioBackend { get; private set; }
+
+        public IEnumerable<AudioBackend> GetPreferredAudioBackendsForCurrentPlatform()
+        {
+            yield return AudioBackend.Automatic;
+
+            switch (RuntimeInfo.OS)
+            {
+                case RuntimeInfo.Platform.Windows:
+                    yield return AudioBackend.Bass;
+
+                    bool wasapiSupported = false;
+
+                    try
+                    {
+                        // Querying device info is the most reliable way to determine if WASAPI is supported.
+                        if (BassWasapi.GetDeviceInfo(BassWasapi.DefaultDevice, out var _) || Bass.LastError != Errors.Wasapi)
+                        {
+                            wasapiSupported = true;
+                        }
+                    }
+                    catch
+                    {
+                        // Ignore any errors in querying bass wasapi devices.
+                    }
+
+                    if (wasapiSupported)
+                    {
+                        // Give preference to WASAPI if available.
```

Comment:
Except as touched on, the offsets are wrong with WASAPI and need further consideration.

Reply:
This seems to be an inaccurate comment. Apologies for the confusion.

```suggestion
                        // Candidate WASAPI if available.
```